config/main.py

```diff
@@ -1893,38 +1892,30 @@ def portchannel(db, ctx, namespace):
 @click.pass_context
 def add_portchannel(ctx, portchannel_name, min_links, fallback):
     """Add port channel"""
-    if is_portchannel_name_valid(portchannel_name) != True:
-        ctx.fail("{} is invalid!, name should have prefix '{}' and suffix '{}'"
-                 .format(portchannel_name, CFG_PORTCHANNEL_PREFIX, CFG_PORTCHANNEL_NO))
-
-    db = ctx.obj['db']
-
+    
     if is_portchannel_present_in_db(db, portchannel_name):
         ctx.fail("{} already exists!".format(portchannel_name))
+    
+    db = ValidatedConfigDBConnector()
+    db.connect()
 
-    fvs = {'admin_status': 'up',
-           'mtu': '9100',
-           'lacp_key': 'auto'}
+    fvs = {"admin_status": "up",
+           "mtu": "9100",  
+           "lacp_key": "auto"}
     if min_links != 0:
-        fvs['min_links'] = str(min_links)
-    if fallback != 'false':
-        fvs['fallback'] = 'true'
+        fvs["min_links"] = str(min_links)
+    if fallback != "false":
+        fvs["fallback"] = "true"
     db.set_entry('PORTCHANNEL', portchannel_name, fvs)
-
+    
 @portchannel.command('del')
 @click.argument('portchannel_name', metavar='<portchannel_name>', required=True)
 @click.pass_context
 def remove_portchannel(ctx, portchannel_name):
     """Remove port channel"""
-    if is_portchannel_name_valid(portchannel_name) != True:
-        ctx.fail("{} is invalid!, name should have prefix '{}' and suffix '{}'"
-                 .format(portchannel_name, CFG_PORTCHANNEL_PREFIX, CFG_PORTCHANNEL_NO))
-
-    db = ctx.obj['db']
```

Comment:
>ctx.obj['db'] [](http://example.com/codeflow?start=9&length=13)

We need to use `ctx.obj['db']` inside ValidatedConfigDBConnector implementation, and the vision is move the decorating to ctx itself, so every place referring `ctx.obj['db']` will benefit with zero code change.